sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/UnsafeRowParquetRecordReader.java

```diff
@@ -716,6 +788,28 @@ private void readLongBatch(int rowId, int num, ColumnVector column) throws IOExc
       }
     }
 
+    private void readFloatBatch(int rowId, int num, ColumnVector column) throws IOException {
+      // This is where we implement support for the valid type conversions.
+      // TODO: implement remaining type conversions
```

Comment:
I'llchange the TODO. I think if the parquet type is float and the spark type is double, we should probably allow that.